src/main/java/com/es/phoneshop/model/product/ProductDao.java

```diff
@@ -3,8 +3,8 @@
 import java.util.List;
 
 public interface ProductDao {
-    Product getProduct(Long id);
+    Product getProduct(Long id) throws ProductNotFoundException;
```

Comment:
please read about Optional

here when we add new service layer, we will call dao which returns Optional<Product>, and then in service we can check and throw your custom ProductNotFoundException in case of no product was found
and we will catch the exception in the servlet for the PDP, because servlet will call service (the 2 task)